### ISSUE TITLE
Window item

### ISSUE THREAD

**Issue by Luke18033** [BEEmod/BEE2-items#1352]
Originally suggested by @TheDarkBomber on [Gitter](https://gitter.im/BenVlodgi/BEE2.4), this would be an item which would allow seeing through a window to the outside of the test chamber.  This would be useful for Overgrown or Old Aperture. If necessary to embed, it should only embed 1 voxel.
- Clean: Glass panels with a BTS area behind them. This would use worldportals.
- Overgrown/Reconstructing: Destroyed panels. On the floor, this would be covered by something, not sure what though.
- Portal 1: Glass panels with a Portal 1-style BTS area behind them. This would use worldportals.
- BTS: Same as Clean, but from the opposite direction (you can see a test chamber through the back of a glass panel).
- Old Aperture: This would replace the Large Observation Grate, which would be changed to an actual observation room. Small observation rooms would still be grates.
- Wheatley Style: Missing panels. Would be covered by a grating when on the floor.
- TWTM: Window showing a BTS area, like in the actual TWTM.
- Blue Portals style: A window showing an outside area with construction materials at the bottom. This is seen in Chamber 07:
![20170107104821_1](https://cloud.githubusercontent.com/assets/19228201/21744274/eedca560-d4c6-11e6-8faf-512955e398bd.jpg)

**Comment by TheDarkBomber** [comment]
I'm already making a Blue Portals style as a separate pack. I do not want it added into the Default Pack.

**Comment by Luke18033** [comment]
A separate pack? I though it was going to be merged into the default pack. Is it because you're not a collaborator?

**Comment by TheDarkBomber** [comment]
No, it's because I want to make separate packs that can match certain 'themes' (e.g. Blue Portals Pack matches Blue Portals).

**Comment by Luke18033** [comment]
That doesn't make much sense...isn't that what packages are for? You have a `blue_portals` package and a `blue_portals_items` package.

**Comment by TheDarkBomber** [comment]
A pack contains packages. The items will need some custom models, (like the Red Flow item, which can be just a red tile.

**Comment by TheDarkBomber** [comment]
Also, the window item may need a max size, like the max size windows from TWTM.

**Comment by Luke18033** [comment]
@Byzarru was the one who requested BP items originally, and he wanted them in the default pack.

**Comment by TheDarkBomber** [comment]
Separate pack now.

**Comment by Luke18033** [comment]
Fine then, I'll just add it into the Default Pack myself. I have a fork of this, I just need to update it to Pr23.

**Comment by TheDarkBomber** [comment]
Nope.

**Comment by Luke18033** [comment]
How do you expect to stop me? You're not even a collaborator.

**Comment by TheDarkBomber** [comment]
:stop_sign: STOP IN THE NAME OF THE LAW. THIS IS PRESIDENT TRUMP. YOU ARE UNDER ARREST AND WILL BE PUT INTO SOLITARY CONFINEMENT. SO STOP. :stop_sign:

**Comment by Luke18033** [comment]
I don't even need to make a fork... 

And no, I'm not stopping.

**Comment by TheDarkBomber** [comment]
You don't have a fork! HA! BLUE PORTALS **SEPARATE** PACK.

**Comment by Luke18033** [comment]
[Yes I do, it's just very outdated.](https://github.com/Luke18033/BEE2-items-Fork)

**Comment by TheDarkBomber** [comment]
I mean it doesn't have Blue Portals. HA!

**Comment by Luke18033** [comment]
I never said I was working on it, just that I will.

**Comment by TheDarkBomber** [comment]
I'm creating the Pack.

**Comment by Luke18033** [comment]
You create the pack, I'll clone it to a branch on this repository. I'll then merge that into `dev` once I have permission from @TeamSpen210.

**Comment by TheDarkBomber** [comment]
You can keep it on the same repository, but on a separate branch, and release it as a separate pack.

**Comment by TeamSpen210** [comment]
You can't really merge from a completely different repro, @TheDarkBomber should just keep it on his own seperate repro.

**Comment by Luke18033** [comment]
Why? It should be part of the default pack.

**Comment by TheDarkBomber** [comment]
It should be separate.

**Comment by Luke18033** [comment]
No it shouldn't.

**Comment by TheDarkBomber** [comment]
Because it's an entirely new set of items.

**Comment by Luke18033** [comment]
So were Rexaura items. That doesn't mean they were made into a separate pack. Just keep it separate for now, once everything is done create a pull request.

@TeamSpen210 I never said I was merging it from another repository...

**Comment by Byzarru** [comment]
Blue Portals items should definitely be with the default pack. Luke, you ninja, I was going to make that exact point.

**Comment by Luke18033** [comment]
So, about the window item...

**Comment by TheDarkBomber** [comment]
@Luke18033 The Rexaura items would have been a separate pack if it wasn't already implemented in TeamSpen210's Addons.

**Comment by Byzarru** [comment]
It shouldn't replace the Obs. Room in any case, but instead be a supplementary aesthetic item. The Old Aperture 'observation room'  is a light source using outside light, while a 'window' would be an observation room proper. In this case, Old Ap Obs Rooms should be renamed Windows and vice versa.

**Comment by TeamSpen210** [comment]
Worldportals are only paractical to use with the Old Aperture grates. Every worldportal requires having a matching room that goes along with it? Additionally, there is a limit of 4 portals/worldportals being on screen at once. If you have move than that, the renderer gets messed up and everything looks weird. So in singleplayer you can have 2 worldportals open at most, and 0 in coop unless you can restrict where players can place their portals (like in the original coop exits, you can't have portals in that room). 

It's ok with the grates because we know there's only one of them. This really needs to use either 3D skybox, or actual space in the level.

**Comment by Luke18033** [comment]
There isn't only one of the grates though if Unlock Default Items is on. I guess that's part of the "weird results" in the stylevar description though. A skybox would probably be better, although you would need to implement a 3D skybox for every style. For Old Aperture Windows, maybe use @Byzarru's suggestion?

**Comment by Luke18033** [comment]
@TheDarkBomber I think a pull request to add BP items to the Default Pack would be the best idea.

**Comment by Byzarru** [comment]
For maintenance areas in P1 and P2, an embedded room should suffice, as well as an observation room in Old Aperture (provided that the window is the same size as the obs. room). The only real need for a worldportal would be in Blue Portals, BTS, and maybe Overgrown, in which the room is larger than 2x1x1.

Oh, and TWTM. I assumed the observation room item would connect to a wall skybox anyways. It's getting special treatment as far as shaping custom observation rooms, so that item would probably be the BTS window while the 'window' becomes the observation room.

**Comment by Luke18033** [comment]
Updated original post.

**Comment by Luke18033** [comment]
Updated original post again.

**Comment by Luke18033** [comment]
I'm going to delete all of the comments related to the BP stuff being separate/merged, they're kinda spam and have nothing to do with the actual suggestion.

**Comment by Luke18033** [comment]
It really can't combine those?

**Comment by Luke18033** [comment]
This would make much more sense with proper chamber exteriors.
